### PR TITLE
RM#47595 : Fix NPE when creating invoice term from move line

### PR DESCRIPTION
**Pull request by aol-axelor**

### CODE REVIEW

**Review by ale-axelor — CHANGES_REQUESTED** [review]
Please review the methods, a service method should not use Context objects as they are specific to the views and must only be called in Controller classes.

**Comment by jmi-axelor** [comment]
Need to apply spotless on your branch

File: axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceTermController.java [hunk 1]
```diff
@@ -110,12 +126,26 @@ public void initInvoiceTermFromInvoice(ActionRequest request, ActionResponse res
           invoiceTermService.initCustomizedInvoiceTerm(invoice, invoiceTerm);
           response.setValues(invoiceTerm);
         } else {
-          MoveLine moveLine = ContextTool.getContextParent(request.getContext(), MoveLine.class, 1);
+          if (request.getContext().getParent().get("_model").toString().contains("MoveLine")) {
```
Comment (jmi-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceTermController.java:129]: Why do you replace the ContexTool.getContextParent?

File: axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceTermController.java [hunk 2]
```diff
@@ -110,12 +126,26 @@ public void initInvoiceTermFromInvoice(ActionRequest request, ActionResponse res
...
+                      .getContext()
+                      .getParent()
+                      .getParent()
+                      .get("_model")
+                      .equals(Move.class.getName())) {
+                move = request.getContext().getParent().getParent().asType(Move.class);
+              }
 
-          if (moveLine != null) {
-            Move move = ContextTool.getContextParent(request.getContext(), Move.class, 2);
```
Comment (jmi-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceTermController.java:116]: Same here

**Pull request merged**